dali/kernels/imgproc/resample/resampling_setup.cc

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cuda_runtime.h>
+#include "dali/kernels/imgproc/resample/resampling_setup.h"
+
+namespace dali {
+namespace kernels {
+
+inline ResamplingFilter GetResamplingFilter(
+    const ResamplingFilters *filters, const FilterDesc &params) noexcept {
+  switch (params.type) {
+    case ResamplingFilterType::Gaussian:
+      return filters->Gaussian(params.radius*0.5f);
+    case ResamplingFilterType::Triangular:
+      return filters->Triangular(params.radius);
+    case ResamplingFilterType::Lanczos3:
+      return filters->Lanczos3();
+  default:
+    return { nullptr, 0, 0, 0 };
+  }
+}
+
+
+void SeparableResamplingSetup::SetupComputation(
+    const TensorListShape<3> &in, const Params &params) {
+  if (!filters)
```

Comment:
In my opinion this function is too long, it's not clear what _should_ it do. I think it should be split apart and with some tests. For me it's impossible to review it in current shape

Reply:
It's not THAT bad. It's less than 100LOC. To split it, I'd have to jump through some hoops like passing around references to the size accumulators or introduce some context object. In the end it would not be simpler at all.

Reply:
OK, I see that I could calculate the offsets afterwards. But that would leave the `sample_descs` in some half-baked stage between the calls, which I don't quite like.